### PR TITLE
security-release-process: add release template

### PR DESCRIPTION
**Pull request by philips**
Add a release template with details on the best practices of explaining
how to identify vulnerability, how to upgrade, and details on the fix.

### CODE REVIEW

**Comment by philips** [comment]
Need to add a mitigation field.

/hold

**Comment by calebamiles** [comment]
@philips,

I believe this makes more sense to add to the release tooling itself, e.g we would improve on the email template that gets sent out with the tooling to include the necessary info rather than creating a template that a release team member would need to manually fill out after cutting the release

**Comment by philips** [comment]
@calebamiles not all things the security team handles is a full Kubernetes release however. For example this was prompted by https://groups.google.com/forum/#!topic/kubernetes-dev/uMqorPLQS4g

File: security-release-process-documentation/email-templates.md [hunk 1]
```diff
@@ -2,6 +2,38 @@
...
+
+Run `$COMPONENT version` and if it says $OLDVERSION or older you are running a vulnerable version.
+
+**How do I upgrade?**
+
+Run `$COMPONENT upgrade` or follow the upgrade instructions at https://kubernetes.io/docs
+
+**Vulnerability Details**
+
+When $COMPONENT $OLDVERSION is ran with default settings it will immediately shut the internet off.
```
Comment (tallclair) [inline security-release-process-documentation/email-templates.md]: nit: s/ran/run/

File: security-release-process-documentation/email-templates.md [hunk 2]
```diff
@@ -2,6 +2,38 @@
...
+
+Hello Kubernetes Community-
+
+A security issue was discovered in versions of $COMPONENT versions $OLDVERSION or older. The issue is $SEVERITY and upgrading to $VERSION of $COMPONENT is encouraged to fix this issue.
+
+**Am I vulnerable?**
+
+Run `$COMPONENT version` and if it says $OLDVERSION or older you are running a vulnerable version.
+
+**How do I upgrade?**
```
Comment (tallclair) [inline security-release-process-documentation/email-templates.md]: Consider adding an optional mitigation section. Sometimes there's a mitigation or workaround that's easier than an upgrade.

**Comment by philips** [comment]
@tallclair I added a mitigation section and addressed the rest of the feedback. Thanks for the review!

Ready for your LGTM.

**Comment by tallclair** [comment]
/lgtm
Thanks!

**Pull request merged**